instrumentation/servlet/servlet-3.0/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/servlet/v3_0/Servlet3OutputStreamWriteBytesAndOffsetAdvice.java

```diff
@@ -0,0 +1,28 @@
+/* SPDX-License-Identifier: Apache-2.0 */
+package io.opentelemetry.javaagent.instrumentation.servlet.v3_0;
+
+import static io.opentelemetry.javaagent.instrumentation.servlet.v3_0.snippet.Injection.getInjectionObject;
+import static io.opentelemetry.javaagent.instrumentation.servlet.v3_0.snippet.Injection.stringInjection;
+
+import io.opentelemetry.javaagent.instrumentation.servlet.v3_0.snippet.InjectedInfo;
+import io.opentelemetry.javaagent.instrumentation.servlet.v3_0.snippet.InjectionObject;
+import java.io.UnsupportedEncodingException;
+import javax.servlet.ServletOutputStream;
+import net.bytebuddy.asm.Advice;
+
+public class Servlet3OutputStreamWriteBytesAndOffsetAdvice {
+
+  @Advice.OnMethodEnter(suppress = Throwable.class)
+  public static void methodEnter(
```

Comment:
I had a thought about an alternative approach that may not be better but let's chat through in case someone asks about this approach on the PR

if you add `skipOn` to the OnMethodEnter annotation, and change the method to return boolean, then we can return `true` by default causing the real method to get invoked, or we can return `false` and have the real method skipped.

we can use this and pass ServletOutputStream to stringInjection and call write on the ServletOutputStream ourselves, returning false/true whether we "handled" the method call or not.

I think this would be a nice approach except for having to deal with the recursive call, when we call `write` I think it will come back into our advice and we would need to use a flag (traditionally a ThreadLocal) to then not end up in an infinite loop

```suggestion
  @Advice.OnMethodEnter(skipOn = Advice.OnDefaultValue.class, suppress = Throwable.class)
  public static boolean methodEnter(
```